src/main/java/com/redhat/devtools/lsp4ij/LSPIJEditorUtils.java

```diff
@@ -0,0 +1,323 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij;
+
+import com.intellij.ide.highlighter.custom.SyntaxTable;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.fileTypes.PlainTextFileType;
+import com.intellij.openapi.fileTypes.impl.AbstractFileType;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.util.CachedValueProvider;
+import com.intellij.psi.util.CachedValuesManager;
+import com.intellij.util.containers.ContainerUtil;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.plugins.textmate.TextMateFileType;
+import org.jetbrains.plugins.textmate.psi.TextMateFile;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Provides utility methods for editor support behaviors.
+ */
+@ApiStatus.Internal
+public final class LSPIJEditorUtils {
+
+    // TODO: Unfortunately the TextMate interface changed in this commit:
```

Comment:
All of this TextMate-related stuff can be uncommented/enabled as soon as the Gradle 2.x-based build work is merged and support for 2023.2 is dropped. I'll be happy to make a pass on that in a single PR when it's ready.